torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -854,23 +852,10 @@ def normalize_shapes(inputs, layout_or_out):
             new_inputs = list(inputs)
             if not is_mkldnn_wgt and isinstance(new_inputs[1], torch.Tensor):
                 if has_free_symbols(view_size):
-                    # If batch size B is dynamic, we need to infer the batch size from the input
+                    # If batch size B is dynamic, we need to set the batch size and possibly stride
                     assert not has_free_symbols(view_size[1:])
-                    size = torch.tensor(new_inputs[1].size()).prod()
-                    fixed_size = torch.tensor(view_size[1:], dtype=torch.int).prod()
-                    eq = sympy.Eq(view_size[0], int(size // fixed_size))
-                    view_size[0] = (size // fixed_size).item()
-
-                    # solve for any free symbols, since they might exist in stride (in HF XLNetLMHeadModel dynamic case)
-                    free_symbols_prod = math.prod(eq.free_symbols)
-                    free_symbols_equivalent = sympy.solve(eq, free_symbols_prod)[0]
-                    if has_free_symbols(view_stride):
-                        for i, sym in enumerate(view_stride):
-                            if has_free_symbols((sym,)):
-                                view_stride[i] = sym.xreplace(
-                                    {free_symbols_prod: free_symbols_equivalent}
-                                )
-
+                    view_size[:] = V.graph.sizevars.size_hints(view_size)
+                    view_stride[:] = V.graph.sizevars.size_hints(view_stride)
```

Comment:
Shall we add guard here?

Reply:
Do you mean to add something like `assert not (has_free_symbols(view_size) or has_free_symbols(view_stride)`? Since `size_hints` always returns ints, there would never be free symbols remaining. 

Or do you mean to add a fallback argument to the calls to size_hints? I was reluctant to do that, since it seems like it's better to fail on the call to size_hints than on the call to as_strided.

Reply:
Since the code is only for constructing the concrete inputs, there is no need for adding the guards.